### PR TITLE
[ci] Add check-runtime-migration job

### PR DESCRIPTION
**Pull request by alvicsam**
The PR adds the `check-runtime-migration` job which runs unit tests for runtime migration (more info here: https://github.com/paritytech/polkadot/pull/5865). The job runs on every PR and checks if the PR has the `E1-runtimemigration` label. If yes - the test is started.

Log of the successful job can be found [here](https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/1750258)

Closes https://github.com/paritytech/ci_cd/issues/573

### CODE REVIEW

**Review by rdovgan — APPROVED** [review]

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -624,6 +624,32 @@ check-try-runtime:
...
+    - job:                         test-node-metrics
+      artifacts:                   false
+  <<:                              *test-pr-refs
+  <<:                              *docker-env
+  <<:                              *compiler-info
+  script:
+    - |
+      export has_runtimemigration_label=$(curl -sS -H "Accept: application/vnd.github+json" \
+        -H "Authorization: token $GITHUB_PR_TOKEN" \
+        https://api.github.com/repos/paritytech/polkadot/issues/$CI_COMMIT_REF_NAME/labels | grep "runtimemigration" | wc -l)
```
Comment (coderobe) [inline .gitlab-ci.yml:642]: perhaps this should use e1 or whatever applicable label code instead of relying on the name not changing, but otherwise lgtm

**Review by coderobe — APPROVED** [review]

**Review by EgorPopelyaev — APPROVED** [review]

File: .gitlab-ci.yml (hunk 1 above)
Comment (alvicsam) [reply .gitlab-ci.yml]: Agree, changed to E1

**Review by sergejparity — APPROVED** [review]

**Review by radupopa2010 — APPROVED** [review]

**Comment by kianenigma** [comment]
THANK YOU SIR. 🎩 ❤️

**Comment by kianenigma** [comment]
against which nodes are you running this?

**Comment by alvicsam** [comment]
It runs on general ci-nodes. Should it be on `bm*` runners?

**Comment by kianenigma** [comment]
Seems to be failing in https://github.com/paritytech/polkadot/pull/5930

**Comment by kianenigma** [comment]
If we can create a daily job that just generates snapshots for all 3 networks for try-runtime, and then use those files in the CI (and download links so developers can use them locally) that's speed up the CI a lot.

**Comment by alvicsam** [comment]
> Seems to be failing in #5930

Yes, but the job itself works correctly but tests failed: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/1853827

**Pull request merged**